src/main/java/storagesystem/model/ReservationHandler.java

```diff
@@ -0,0 +1,149 @@
+package storagesystem.model;
+
+import org.joda.time.Interval;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A handler taking care of all reservations for one organisation. Methods for creating and retrieving reservations
+ */
+
+public class ReservationHandler {
+
+
+    private List<IReservation> reservations;
+
+
+    public ReservationHandler(int nextReservationID, List<IReservation> reservations) {
+        Reservation.setNxtResId(nextReservationID);
+        this.reservations = reservations;
+    }
+
+
+    /**
+     * @param id
+     * @return Reservation matching id. Returns null if none found.
+     */
+
+    public IReservation getReservation(int id) throws NullPointerException {
+        for (IReservation res : reservations) {
+            if (res.getID() == id) {
+                return res.copy();
+            }
+        }
+        return null;
+    }
+
+
+    /**
+     * @return List with all known reservations
+     */
+    public List<IReservation> getReservations() {
+        return reservations;
+    }
+
+    /**
+     * @param object
+     * @return A list with all known reservations with specific object.
+     */
+    public List<IReservation> getReservations(IReservable object) {
+
+
+        List<IReservation> reservations = new ArrayList<>();
```

Comment:
Namechange for clarity throughout this method

```suggestion
        List<IReservation> objectsReservations = new ArrayList<>();
```